test/functional/test_framework/util.py

```diff
@@ -208,10 +208,13 @@ def str_to_b64str(string):
 def satoshi_round(amount):
     return Decimal(amount).quantize(Decimal('0.00000001'), rounding=ROUND_DOWN)
 
-def wait_until(predicate, *, attempts=float('inf'), timeout=float('inf'), lock=None, factor=1.0):
-    if attempts == float('inf') and timeout == float('inf'):
+def wait_until(predicate, *, attempts=float('inf'), timeout=float('inf'), lock=None, timeout_factor=1.0):
+    if timeout_factor == 0:
```

Comment:
Why is this needed? The condition should never be True.

Reply:
This was required because I was setting the factor to 0. After the above changes it's not required anymore. Reverted.